contracts/v2/sovereignChains/BridgeL2SovereignChain.sol

```diff
@@ -378,18 +360,76 @@ contract BridgeL2SovereignChain is
         );
     }
 
+    /**
+     * @notice unset multiple claims from the claimedBitmap
+     * @dev This function is a "multi/batch call" to `unsetClaimedBitmap`
+     * @param leafIndexes Array of Index
+     * @param sourceBridgeNetworks Array of Origin networks
+     */
+    function unsetMultipleClaimedBitmap(
+        uint32[] memory leafIndexes,
+        uint32[] memory sourceBridgeNetworks
+    ) external onlyBridgeManager {
+        if (leafIndexes.length != sourceBridgeNetworks.length) {
+            revert InputArraysLengthMismatch();
+        }
+
+        for (uint256 i = 0; i < leafIndexes.length; i++) {
+            _unsetClaimedBitmap(leafIndexes[i], sourceBridgeNetworks[i]);
+        }
+    }
+
     /**
      * @notice Updated bridge manager address
      * @param _bridgeManager Bridge manager address
      */
     function setBridgeManager(
         address _bridgeManager
     ) external onlyBridgeManager {
-        if (_bridgeManager == address(0)) revert NotValidBridgeManager();
         bridgeManager = _bridgeManager;
         emit SetBridgeManager(bridgeManager);
     }
 
+    /**
+     * @notice Function to check if an index is claimed or not
+     * @dev function override to improve a bit the performance and bytecode not checking unnecessary conditions for sovereign chains context
+     * @param leafIndex Index
+     * @param sourceBridgeNetwork Origin network
+     */
+    function isClaimed(
+        uint32 leafIndex,
+        uint32 sourceBridgeNetwork
+    ) external view override returns (bool) {
+        uint256 globalIndex = uint256(leafIndex) +
+            uint256(sourceBridgeNetwork) *
+            _MAX_LEAFS_PER_NETWORK;
+
+        (uint256 wordPos, uint256 bitPos) = _bitmapPositions(globalIndex);
+        uint256 mask = (1 << bitPos);
+        return (claimedBitMap[wordPos] & mask) == mask;
+    }
+
+    /**
+     * @notice Function to check that an index is not claimed and set it as claimed
+     * @dev function override to improve a bit the performance and bytecode not checking unnecessary conditions for sovereign chains context
+     * @param leafIndex Index
+     * @param sourceBridgeNetwork Origin network
+     */
+    function _setAndCheckClaimed(
+        uint32 leafIndex,
+        uint32 sourceBridgeNetwork
+    ) internal override {
+        uint256 globalIndex = uint256(leafIndex) +
```

Comment:
join ( in the same SC section) and add comments for all this functions that are overrided to save bytecode

Reply:
The comment is added at the @dev tag of all those functions that are being overrided